containers/src/Data/IntMap/Internal.hs

```diff
@@ -3473,12 +3474,74 @@ instance Eq1 IntMap where
 --------------------------------------------------------------------}
 
 instance Ord a => Ord (IntMap a) where
-    compare m1 m2 = compare (toList m1) (toList m2)
+  compare m1 m2 = liftCmp compare m1 m2
+  {-# INLINABLE compare #-}
 
 -- | @since 0.5.9
 instance Ord1 IntMap where
-  liftCompare cmp m n =
-    liftCompare (liftCompare cmp) (toList m) (toList n)
+  liftCompare = liftCmp
+
+liftCmp :: (a -> b -> Ordering) -> IntMap a -> IntMap b -> Ordering
+liftCmp cmp m1 m2 = case (splitSign m1, splitSign m2) of
+  ((l1, r1), (l2, r2)) -> case go l1 l2 of
+    Less -> LT
+    Prefix' -> if null r1 then LT else GT
+    Equals -> case go r1 r2 of
```

Comment:
Consider moving this case to the bottom for readability.

Reply:
Do you think that would be better? I prefer the consistent order of branches.

Reply:
I don't feel super strongly about it.